src/shared/components/post/post-listing.tsx

```diff
@@ -101,14 +132,146 @@ export class PostListing extends Component<PostListingProps, PostListingState> {
     });
   }
 
+  componentDidUpdate(
```

Comment:
What is going on here?

`isFocused` should be a prop. 

I don't see why you need a masonry updates, we're already doing that elsewhere.

isHighlighted should be a prop and I don't see why a masonry update is necessary.